### PR TITLE
Make Breakdown limit customizable and Allow empty breakdown value in trends and funnels

### LINKED ISSUE #5341: Inconsistency between breakdown and roll-up of funnel
## Bug description
This is debatable as to whether its a bug - but I think it could cause a lot of confusion and loss of trust in our product.

I create a funnel and can see 1,875 were successful:
image

Then I break down by browser and the number is now only 1,746
image

## Expected behavior
I would assume we would show the same number in total from a breakdown or a non-broken down funnel step.

We may be omitting an "other" category which catches the smaller things we didn't break down by?

I'm also curious what's limiting us from having more breakdowns? Could we get 10 rather than limiting to what appears to be 5 today?

## How to reproduce

1. Take any funnel step with a significant number of events and break it down by browser
2. https://app.posthog.com/insights?insight=FUNNELS&properties=%5B%5D&filter_test_accounts=true&events=%5B%7B%22id%22%3A%22%24pageview%22%2C%22name%22%3A%22%24pageview%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%7D%2C%7B%22id%22%3A%22%24pageview%22%2C%22name%22%3A%22%24pageview%22%2C%22type%22%3A%22events%22%2C%22order%22%3A1%7D%2C%7B%22id%22%3A%22%24pageview%22%2C%22name%22%3A%22%24pageview%22%2C%22type%22%3A%22events%22%2C%22order%22%3A2%7D%5D&actions=%5B%5D&interval=day&new_entity=%5B%5D&funnel_viz_type=steps&display=FunnelViz&date_from=-1d&date_to=dStart&breakdown=%24browser&breakdown_type=event
3.

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: _please provide_

## Additional context
cc: @EDsCODE @alexkim205 @macobo @neilkakkar  as this is likely to requires some core experience / core analytics collaboration.

cc: @clarkus  for any design considerations here

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

Adds a customizable limit to the number of breakdown values. Resolves https://github.com/PostHog/posthog/issues/5341

Also does the same for trends, and changes how `none` breakdown used to work: Now, none isn't returned if it's empty, and is treated just like any other breakdown value (so works with limit/offset as well)

For the frontend: Since now `''` is a valid breakdown value, I update the labels to work with this (and added tests for this)

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Comment by neilkakkar** [comment]
Labels still broken for Funnels, fixing that

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelStepTable.tsx [hunk 1]
```diff
@@ -51,14 +51,16 @@ export function FunnelStepTable({}: FunnelStepTableProps): JSX.Element | null {
         render: function RenderLabel({}, step: FlattenedFunnelStep): JSX.Element {
             const isBreakdownChild = !!filters.breakdown && !step.isBreakdownParent
             const color = getStepColor(step, !!filters.breakdown)
+            console.log(step)
```
Comment (neilkakkar) [inline frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelStepTable.tsx:54]: oops, rm!

**Review by alexkim205 — APPROVED** [review]

File: ee/clickhouse/sql/trends/top_elements.py [hunk 2]
```diff
@@ -6,7 +6,6 @@
     FROM events e
     WHERE
         team_id = %(team_id)s {entity_query} {parsed_date_from} {parsed_date_to} {prop_filters}
-     AND JSONHas(properties, %(key)s)
```
Comment (EDsCODE) [inline ee/clickhouse/sql/trends/top_elements.py:9]: Oh, was this what was preventing none results from returning? I think that may have led me to implement the workaround for unioning the events with no property

**Review by EDsCODE — APPROVED** [review]
backend changes look good

File: ee/clickhouse/sql/trends/top_elements.py (hunk 2 above)
Comment (neilkakkar) [reply ee/clickhouse/sql/trends/top_elements.py:9]: correct!

**Pull request merged**